docs/topics/build-tools-api.md

```diff
@@ -0,0 +1,111 @@
+a unified API between build systems and the Kotlin compiler ecosystem.
+
+The BTA defines a single entry point that build systems can implement. This removes the need to deeply integrate with internal compiler details.
+
+Currently, the BTA supports Kotlin/JVM **only**.
+
+> The BTA itself is not yet publicly available for direct use in your own build tool integrations.
+> If you're interested in the proposal or want to share feedback, see the [KEEP](https://github.com/Kotlin/KEEP/issues/421).
+> Follow the status of its implementation in [YouTrack](https://youtrack.jetbrains.com/issue/KT-76255).
+> 
+{style="warning"}
+
+## Integration with Gradle
+
+<primary-label ref="experimental-general"/>
+
+The Kotlin Gradle plugin (KGP) already supports the BTA.
+
+### How to enable
+
+Add the following property to your `gradle.properties` file:
+
+```kotlin
+kotlin.compiler.runViaBuildToolsApi=true
+```
+
+### Configure different compiler versions
+
+With the BTA, you can now use a different Kotlin compiler version than the version used by the KGP. This is useful when:
+
+* You want to try new Kotlin features but haven't updated your build scripts yet.
+* You need the latest plugin fixes but want to stay on an older compiler version for now.
+
+Here's an example of how to configure this in your `build.gradle.kts` file:
+
+```kotlin
+import org.jetbrains.kotlin.buildtools.api.ExperimentalBuildToolsApi
+import org.jetbrains.kotlin.gradle.ExperimentalKotlinGradlePluginApi
+
+plugins {
+	kotlin("jvm") version "2.2.0"
+}
+
+group = "org.jetbrains.example"
+version = "1.0-SNAPSHOT"
+
+repositories {
+	mavenCentral()
+}
+
+kotlin {
+	jvmToolchain(8)
+	@OptIn(ExperimentalBuildToolsApi::class, ExperimentalKotlinGradlePluginApi::class)
+	compilerVersion.set("2.1.21") // <-- different version than 2.2.0
+}
+```
+
+#### Compatible Kotlin compiler and KGP versions
+
+The BTA supports:
+
+* The three previous major Kotlin compiler versions.
+* One major version forward.
+
+For example, in KGP 2.2.0, the supported Kotlin compiler versions are:
+
+* 2.1.x
+* 2.0.x 
+* 1.9.25
+* Forward-compatible with 2.2.x and 2.3.x
+
+#### Limitations
+
+Using different compiler versions together with compiler plugins may lead to Kotlin compiler exceptions. The Kotlin team
+plans to address this in future Kotlin releases.
+
+### Enable incremental compilation with "in process" compiler execution strategy
+
+The KGP supports three [compiler execution strategies](gradle-compilation-and-caches.md#defining-kotlin-compiler-execution-strategy).
+Ordinarily, the "in-process" strategy (which runs the compiler in the Gradle daemon) doesn't support incremental compilation.
+
+With the BTA, the "in-process" strategy now supports incremental compilation. To enable it, add the following property to
+your `gradle.properties` file:
+
+```kotlin
+kotlin.compiler.execution.strategy=in-process
+```
+
+### Share feedback
```

Comment:
This actually could be a What's next since this page doesn't have it yet (plus this way you have fewer sections on the right side for the main text)

Alternatively, this could be a line in the red warning at the top, since we ask for feedback there as well.

Reply:
Well, since BTA doesn't do much Maven users, I think it's totally fine to have it the general warning

Reply:
I see. What about dropping the header here? we can just leave a simple paragraph or make a tip out of this. I think 1 sentence doesn't deserve a separate section here